tests/unit/python/fledge/services/core/api/plugins/test_remove.py

```diff
@@ -0,0 +1,261 @@
+        storage_client_mock = MagicMock(StorageClientAsync)
+        plugin_installed = [{"name": "sinusoid", "type": _type, "description": "Sinusoid Poll Plugin",
+                             "version": "1.8.1", "installedDirectory": "{}/{}".format(_type, name),
+                             "packageName": "fledge-{}-sinusoid".format(_type)},
+                            {"name": name, "type": _type, "description": "HTTP Listener South Plugin",
+                             "version": "1.8.1", "installedDirectory": "{}/{}".format(_type, name),
+                             "packageName": "fledge-{}-{}".format(_type, name)}
+                            ]
+        with patch.object(PluginDiscovery, 'get_plugins_installed', return_value=plugin_installed
+                          ) as plugin_installed_patch:
+            with patch.object(plugins_remove, '_check_plugin_usage', return_value=async_mock([])) as plugin_usage_patch:
+                with patch.object(plugins_remove._logger, "info") as log_info_patch:
+                    with patch.object(connect, 'get_storage_async', return_value=storage_client_mock):
+                        with patch.object(storage_client_mock, 'query_tbl_with_payload',
+                                          return_value=async_mock(select_row_resp)) as query_tbl_patch:
+                            resp = await client.delete('/fledge/plugins/{}/{}'.format(_type, name), data=None)
+                            assert 429 == resp.status
+                            assert expected_msg == resp.reason
+                            r = await resp.text()
+                            actual = json.loads(r)
+                            assert {'message': expected_msg} == actual
+                        args, kwargs = query_tbl_patch.call_args_list[0]
+                        assert 'packages' == args[0]
+                        assert payload == json.loads(args[1])
+                assert 1 == log_info_patch.call_count
+                log_info_patch.assert_called_once_with(
+                    'No entry found for http_south plugin in asset tracker; '
+                    'or {} plugin may have been added in disabled state & never used'.format(name))
+            plugin_usage_patch.assert_called_once_with(_type, name)
+        plugin_installed_patch.assert_called_once_with(_type, False)
+
+    async def test_package_when_not_in_use(self, client):
+        async def async_mock(return_value):
+            return return_value
+        _type = "south"
+        name = 'http_south'
+        pkg_name = "fledge-south-http-south"
+        payload = {"return": ["status"], "where": {"column": "action", "condition": "=", "value": "purge",
+                                                   "and": {"column": "name", "condition": "=", "value": pkg_name}}}
+        select_row_resp = {'count': 1, 'rows': [{
+            "id": "c5648940-31ec-4f78-a7a5-b1707e8fe578",
```

Comment:
this for last failure call check?  ID does not matter as patched here, but should be unique on new request insert

Reply:
This is the return value (And expected one). For return values as close as with real storage result.